perf/src/discard.rs

```diff
@@ -0,0 +1,34 @@
+use {
+    crate::packet::PacketBatch,
+    rand::{thread_rng, Rng},
+};
+
+pub fn discard_batches_randomly(
+    batches: &mut Vec<PacketBatch>,
```

Comment:
This is essentially a random sample. Could you try something like this API and see how it compares? https://docs.rs/rand/latest/rand/seq/index/fn.sample.html

Reply:
So it would have to allocate and return a vector of indexes? then do a retain with the result? Seems like it's going to be slow.

Reply:
Yeah, you're right, never mind